src/BoundingBox.h

```diff
@@ -1,8 +1,10 @@
 #ifndef BOUNDING_BOX_H
 #define BOUNDING_BOX_H
 
-#define EIGEN_DONT_ALIGN_STATICALLY
-#include <Eigen/Dense>
+#include "glm/glm.hpp"
+#include "glm/vec4.hpp"

```

Comment:
Is `vec3` included in `glm/vec4.hpp`? I haven't seen this header, we sure it's necessary?